JBang could not find dependency `net.sf.saxon:Saxon-HE:11.4`

**Describe the bug**
JBang 0.109.0 is unable to fetch the dependency `net.sf.saxon:Saxon-HE:11.4`, raising a `ResourceNotFoundException`.

**To Reproduce**
Steps to reproduce the behavior:
1. Have an script with the following dependencies:
```
///usr/bin/env jbang "$0" "$@" ; exit $?

//JAVA 17+

//DEPS net.sf.saxon:Saxon-HE:11.4

class Hello {

    public static void main(final String... args) { 
        System.out.println("Hello JBang");
     }
}
```

2.  Start a fedora:36 (or fedora:38) container, mounting the script (or making it available inside the container somehow).
```
podman run -ti --entrypoint /bin/sh -v $(pwd):/tmp/coursebook:z registry.fedoraproject.org/fedora:36-x86_64
```
3. Install JBang:
```
> curl -Ls https://sh.jbang.dev | JBANG_DOWNLOAD_VERSION=0.109.0 bash -s - app setup
Downloading JBang 0.109.0...
Installing JBang...
Downloading JDK 17. Be patient, this can take several minutes...
Installing JDK 17...
[jbang] Default JDK set to 17 (17.0.7, current, /root/.jbang/cache/jdks/17)
[jbang] Setting up JBang environment...
Please start a new Shell for changes to take effect
> bash
[root@5a8dd007075a /]# 
```
3. run the script
```
[root@5a8dd007075a /]# cd /tmp/coursebook
[root@5a8dd007075a coursebook]# jbang --verbose run Hello.java
```

5. See error
```
[jbang] [0:193] jbang version 0.109.0
[jbang] [0:203] Resolving resource ref: Hello.java
[jbang] [0:206] Resolved resource ref as: Hello.java (cached as: /tmp/coursebook/Hello.java)
[jbang] [0:224] [ERROR] Could not find net.sf.saxon:Saxon-HE:11.4
dev.jbang.cli.ResourceNotFoundException: Could not find net.sf.saxon:Saxon-HE:11.4
        at dev.jbang.source.resolvers.SiblingResourceResolver.resolve(SiblingResourceResolver.java:55)
        at dev.jbang.source.ProjectBuilder.updateProject(ProjectBuilder.java:512)
        at dev.jbang.source.ProjectBuilder.updateProjectMain(ProjectBuilder.java:468)
        at dev.jbang.source.ProjectBuilder.createSourceProject(ProjectBuilder.java:341)
        at dev.jbang.source.ProjectBuilder.build(ProjectBuilder.java:269)
        at dev.jbang.source.ProjectBuilder.build(ProjectBuilder.java:218)
        at dev.jbang.cli.Run.doCall(Run.java:73)
        at dev.jbang.cli.BaseCommand.call(BaseCommand.java:145)
        at dev.jbang.cli.BaseCommand.call(BaseCommand.java:21)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
        at picocli.CommandLine.access$1300(CommandLine.java:145)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2358)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2352)
        at dev.jbang.cli.JBang$3.handle(JBang.java:148)
        at dev.jbang.cli.JBang$3.handle(JBang.java:143)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2316)
        at picocli.CommandLine.execute(CommandLine.java:2078)
        at dev.jbang.Main.main(Main.java:14)
[jbang] [0:224] If you believe this a bug in jbang, open an issue at https://github.com/jbangdev/jbang/issues
```

**Expected behavior**
Dependencies fetched and script ran succesfully.

**JBang version**
[jbang] [0:194] jbang version 0.109.0
Cache: /home/jordisola/.jbang/cache
Config: /home/jordisola/.jbang
Repository:/home/jordisola/.m2/repository
0.109.0

**Additional context**
JBang 0.108.0 works fine following the same steps.

--- a/src/main/java/dev/jbang/dependencies/DependencyUtil.java
+++ b/src/main/java/dev/jbang/dependencies/DependencyUtil.java
@@ -44,10 +44,10 @@ public class DependencyUtil {
 	}
 
 	public static final Pattern fullGavPattern = Pattern.compile(
-			"^(?<groupid>[a-z0-9_.-]*):(?<artifactid>[a-z0-9_.-]*):(?<version>[^:@]*)(:(?<classifier>[^@]*))?(@(?<type>.*))?$");
+			"^(?<groupid>[a-zA-Z0-9_.-]*):(?<artifactid>[a-zA-Z0-9_.-]*):(?<version>[^:@]*)(:(?<classifier>[^@]*))?(@(?<type>.*))?$");
 
 	public static final Pattern lenientGavPattern = Pattern.compile(
-			"^(?<groupid>[a-z0-9_.-]*):(?<artifactid>[a-z0-9_.-]*)(:(?<version>[^:@]*)(:(?<classifier>[^@]*))?)?(@(?<type>.*))?$");
+			"^(?<groupid>[a-zA-Z0-9_.-]*):(?<artifactid>[a-zA-Z0-9_.-]*)(:(?<version>[^:@]*)(:(?<classifier>[^@]*))?)?(@(?<type>.*))?$");
 
 	private DependencyUtil() {
 	}

--- a/src/test/java/dev/jbang/source/TestTagReader.java
+++ b/src/test/java/dev/jbang/source/TestTagReader.java
@@ -15,10 +15,10 @@ public class TestTagReader {
 	@Test
 	void testExtractDependencies() {
 		TagReader tr = new TagReader.Extended(
-				"//DEPS foo:bar, abc:def:123, https://github.com/jbangdev/jbang, something", null);
+				"//DEPS foo:bar, abc:DEF:123, https://github.com/jbangdev/jbang, something", null);
 
 		List<String> deps = tr.collectBinaryDependencies();
-		assertThat(deps, containsInAnyOrder("foo:bar", "abc:def:123", "https://github.com/jbangdev/jbang"));
+		assertThat(deps, containsInAnyOrder("foo:bar", "abc:DEF:123", "https://github.com/jbangdev/jbang"));
 
 		List<String> subs = tr.collectSourceDependencies();
 		assertThat(subs, containsInAnyOrder("something"));

